docs/sections/learn/steps/index.md

```diff
@@ -0,0 +1,134 @@
+
+## An example: ConversationTemplate
+
+Let's see one simple type of step as an example, the [`ConversationTemplate`][distilabel.steps.conversation.ConversationTemplate]. Let's take a look at it's definition (the docstrings are removed for clarity, but it can be reviewd in the API reference):
+
+```python
+class ConversationTemplate(Step):
+
+    @property
+    def inputs(self) -> List[str]:
+        return ["instruction", "response"]
+
+    @property
+    def outputs(self) -> List[str]:
+        return ["conversation"]
+
+    def process(self, inputs: StepInput) -> "StepOutput":
+        for input in inputs:
+            input["conversation"] = [
+                {"role": "user", "content": input["instruction"]},
+                {"role": "assistant", "content": input["response"]},
+            ]
+        yield inputs
+```
+
+At the very minimal, we need to define the `inputs` and `outputs` properties with the column names required as input, and returned as output respectively, and the processing logic of the step in the `process` method.
+
+In this example, we see that it takes `inputs` as argument, annotated as `StepInput`, which is a list of dictionaries with the data, and *yields* a `StepOutput`.
+
+### Working with the step
+
+Let's see how to instantiate this `Step` outside of a `Pipeline`:
+
+```python
+from distilabel.pipeline.local import Pipeline
+from distilabel.steps.conversation import ConversationTemplate
+
+conversation_template = ConversationTemplate(
+    name="my-conversation-template",
+    pipeline=Pipeline(name="my-pipeline"),
+)
+```
+
+As we mentioned, every `Step` must be defined in the context of a `Pipeline`, which means we need to pass it as an argument if we decide to create a standalone step. If we take a look at the `conversation_template` step, we see the following fields:
+
+```python
+conversation_template
+# ConversationTemplate(name='my-conversation-template', input_mappings={}, output_mappings={}, input_batch_size=50)
+```
+
+The `name` of the `Step`, a mandatory field to identify the `Step` within the `Pipeline`. `input_mappings`, which is a dictionary that can be useful when the names of the output columns of the current step differ from the names of the input columns of the following step/s, and `output_mappings` which can be used to verify the output columns of this step exist after processing the data. If this two fields are left to the default (without content), the different steps will assume the column names along the different steps will be present, and raise an error if that's not the case. And `input_batch_size` (by default set to 50), which is independent for every step and will determine how often it can be executed. If won't matter that much in this step, but as we will see later, other types of steps will come with an `LLM`, so having this flexibility will be really useful.
```

Comment:
```suggestion
- The `name` of the `Step`, a mandatory field to identify the `Step` within the `Pipeline`. 
- `input_mappings`, which is a dictionary that can be useful to map keys from the input dictionaries to the keys expected by the step. For example, if `input_mappings={"instruction": "prompt"}`, that means that the key `prompt` from the input dictionaries will be used as the key `instruction` for the step.
- `output_mappings`, which is a dictionary that can be used to map the `outputs` of the step to other names. For example, if `output_mappings={"conversation": "prompt"}`, that means that the key `conversation` generated by the step will be renamed to `prompt` and the output dictionaries of this step will contain a key called `prompt` instead of `conversation`.
- `input_batch_size` (by default set to 50), which is independent for every step and will determine how many input dictionaries will process at once. If won't matter that much in this step, but as we will see later, other types of steps will come with an `LLM`, so having this flexibility will be really useful.
```